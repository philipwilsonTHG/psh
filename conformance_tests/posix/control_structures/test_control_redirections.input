# Test redirections on control structures
# PSH supports I/O redirection on control structures (v0.23.0 feature)

mkdir -p tmp

# Test while loop with output redirection
while read line; do
    echo "Processed: $line"
done << EOF > tmp/while_output.txt
line1
line2
line3
EOF

echo "While loop output:"
cat tmp/while_output.txt

# Test for loop with redirection
for item in apple banana cherry; do
    echo "Item: $item"
done > tmp/for_output.txt

echo "For loop output:"
cat tmp/for_output.txt

# Test if statement with redirection
if true; then
    echo "True branch output"
    echo "Second line"
fi > tmp/if_output.txt

echo "If statement output:"
cat tmp/if_output.txt

# Test case statement with redirection
value="test"
case "$value" in
    test)
        echo "Case matched: $value"
        echo "Additional output"
        ;;
    *)
        echo "No match"
        ;;
esac > tmp/case_output.txt

echo "Case statement output:"
cat tmp/case_output.txt

# Test nested redirection
for num in 1 2; do
    echo "Number: $num"
    if [ "$num" = "2" ]; then
        echo "Special number: $num"
    fi
done > tmp/nested_output.txt

echo "Nested redirection output:"
cat tmp/nested_output.txt

# Cleanup
rm -rf tmp