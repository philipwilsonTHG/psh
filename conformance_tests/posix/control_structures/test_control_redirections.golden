While loop output:
For loop output:
Item: apple
Item: banana
Item: cherry
If statement output:
True branch output
Second line
Case statement output:
Case matched: test
Additional output
Nested redirection output:
Number: 1
Number: 2
Special number: 2
